crates/flux-middle/src/rustc/ty.rs

```diff
@@ -182,6 +182,20 @@ pub enum TyKind {
     CoroutineWitness(DefId, GenericArgs),
     Alias(AliasKind, AliasTy),
     RawPtr(Ty, Mutability),
+    TraitObject(List<PolyTraitRef>, Region, TraitObjectSyntax),
```

Comment:
This is the wrong representation at this level, instead, it should match [`TyKind::Dynamic`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/type.TyKind.html#variant.Dynamic).